espnet2/sds/espnet_model.py

```diff
@@ -0,0 +1,256 @@
+        super().__init__()
+        self.TTS_option = TTS_option
+        self.ASR_option = ASR_option
+        self.LLM_option = LLM_option
+        self.type_option = type_option
+        self.access_token = access_token
+        self.TTS_curr_name = None
+        self.LLM_curr_name = None
+        self.ASR_curr_name = None
+        self.text2speech = None
+        self.s2t = None
+        self.LM_pipe = None
+        self.client = None
+        self.vad_model = WebrtcVADModel()
+        self.chat = Chat(2)
+        self.chat.init_chat(
+            {
+                "role": "system",
+                "content": (
+                    "You are a helpful and friendly AI "
+                    "assistant. "
+                    "You are polite, respectful, and aim to "
+                    "provide concise and complete responses of "
+                    "less than 15 words."
+                ),
+            }
+        )
+        self.user_role = "user"
+
+    def handle_TTS_selection(self, option):
+        if self.TTS_curr_name is not None:
+            if option == self.TTS_curr_name:
+                return
+        yield gr.Textbox(visible=False), gr.Textbox(visible=False), gr.Audio(
+            visible=False
+        )
+        self.TTS_curr_name = option
+        tag = option
+        if tag == "ChatTTS":
+            self.text2speech = ChatTTSModel()
+        else:
+            self.text2speech = ESPnetTTSModel(tag)
+        self.text2speech.warmup()
+        yield gr.Textbox(visible=True), gr.Textbox(visible=True), gr.Audio(visible=True)
+
+    def handle_LLM_selection(self, option):
+        if self.LLM_curr_name is not None:
+            if option == self.LLM_curr_name:
+                return
+        yield gr.Textbox(visible=False), gr.Textbox(visible=False), gr.Audio(
+            visible=False
+        )
+        self.LLM_curr_name = option
+        self.LM_pipe = HuggingFaceLLM(access_token=self.access_token, tag=option)
+        self.LM_pipe.warmup()
+        yield gr.Textbox(visible=True), gr.Textbox(visible=True), gr.Audio(visible=True)
+
+    def handle_ASR_selection(self, option):
+        if option == "librispeech_asr":
+            option = (
+                "espnet/"
+                "simpleoier_librispeech_asr_train_asr_conformer7"
+                "_wavlm_large_raw_en_bpe5000_sp"
+            )
+        if self.ASR_curr_name is not None:
+            if option == self.ASR_curr_name:
+                return
+        yield gr.Textbox(visible=False), gr.Textbox(visible=False), gr.Audio(
+            visible=False
+        )
+        self.ASR_curr_name = option
+        if option == "espnet/owsm_v3.1_ebf":
+            self.s2t = OWSMModel()
+        elif option == (
+            "espnet/"
+            "simpleoier_librispeech_asr_train_asr_conformer7"
+            "_wavlm_large_raw_en_bpe5000_sp"
+        ):
+            self.s2t = ESPnetASRModel(tag=option)
+        elif option == "whisper":
```

Comment:
it will be better if you replace with something like:
```
elif "whisper" in option:
            self.s2t = WhisperASRModel(tag=option.replace("whisper-", ""))
```
to handle other whisper models.

Reply:
Sounds great! I have incorporated your suggestion!!